libdnf/module/ModulePackage.cpp

```diff
@@ -0,0 +1,181 @@
+#include "ModulePackage.hpp"
+#include "libdnf/module/state/EnabledState.hpp"
+
+#include <sstream>
+#include <utility>
+#include <iostream>
+
+ModulePackage::ModulePackage(Pool *pool, Repo *repo, const std::shared_ptr<ModuleMetadata> &metadata)
+        : metadata(metadata)
+        , state(std::make_shared<ModuleState>())
+        , pool(pool)
+{
+    id = repo_add_solvable(repo);
+    Solvable *solvable = pool_id2solvable(pool, id);
+
+    createProvisions(solvable);
+    createDependencies(solvable);
+}
+
+void ModulePackage::createProvisions(Solvable *solvable) const
+{
+    std::ostringstream ss;
+    ss << getName() << "-" << getStream() << "-" << getVersion() << "-" << getContext();
+    solvable_set_str(solvable, SOLVABLE_NAME, ss.str().c_str());
+    solvable_set_str(solvable, SOLVABLE_EVR, "0");
+    solvable_set_str(solvable, SOLVABLE_ARCH, metadata->getArchitecture().c_str());
+
+    ss.str(std::string());
+    ss << "module-n(" << getName() << ")";
+    auto depId = pool_str2id(pool, ss.str().c_str(), 1);
+    solvable_add_deparray(solvable, SOLVABLE_PROVIDES, depId, -1);
+
+    ss.str(std::string());
+    ss << "module-ns(" << getNameStream() << ")";
+    depId = pool_str2id(pool, ss.str().c_str(), 1);
+    solvable_add_deparray(solvable, SOLVABLE_PROVIDES, depId, -1);
+
+    ss.str(std::string());
+    ss << "module-nsv(" << getNameStreamVersion() << ")";
+    depId = pool_str2id(pool, ss.str().c_str(), 1);
+    solvable_add_deparray(solvable, SOLVABLE_PROVIDES, depId, -1);
+}
+
+void ModulePackage::createDependencies(Solvable *solvable) const
+{
+    std::ostringstream ss;
+    Id depId;
+    
+    for (const auto &dependency : getModuleDependencies()) {
+        for (const auto &requires : dependency->getRequires()) {
+            for (const auto &singleRequires : requires) {
+                auto moduleName = singleRequires.first;
+
+                bool hasStreamRequires = false;
+                for (const auto &moduleStream : singleRequires.second) {
+                    ss.str(std::string());
+                    ss << "module-ns(" << moduleName;
+                    if (moduleStream.find('-', 0) != std::string::npos) {
+                        ss << ":" << moduleStream.substr(1) << ")";
+                        depId = pool_str2id(pool, ss.str().c_str(), 1);
+                        solvable_add_deparray(solvable, SOLVABLE_CONFLICTS, depId, -1);
```

Comment:
no, it doesn't conflict. please look how it is done properly with rich dependencies in fus or fm-orchestrator